lib/bacnet-asn1.js

```diff
@@ -699,11 +700,15 @@ var bacapp_decode_application_data = module.exports.bacapp_decode_application_da
       var len = result.len;
       result = bacapp_decode_data(buffer, offset + len, maxOffset, result.tagNumber, result.value);
       if (!result) return;
-      return {
+      var resObj = {
         len: len + result.len,
         type: result.type,
         value: result.value
       };
+      if (result.encoding !== undefined) {
```

Comment:
Well, in this specific case it is decoding, and here we only have to validate what we set in our code. And encoding is either set to an integer or not set at all (`undefined`). So we don't have to check for `null` here.